test/rebar_compile_SUITE.erl

```diff
@@ -1531,6 +1531,22 @@ override_deps(Config) ->
               {dep_not_exist, "other_dep"}]}
     ).
 
+git_subdir_deps(Config) ->
+    Deps = rebar_test_utils:expand_deps(git_subdir, [{"some_dep", "0.0.1", [{"other_dep", "0.0.1", []}]}]),
+    TopDeps = rebar_test_utils:top_level_deps(Deps),
+
+    {SrcDeps, _} = rebar_test_utils:flat_deps(Deps),
+    mock_git_subdir_resource:mock([{deps, SrcDeps}]),
+
+    RebarConfig = [
+        {deps, TopDeps}
+    ],
+    rebar_test_utils:run_and_check(
+        Config, RebarConfig, ["compile"],
+        {ok, [{subdir_dep, "some_dep"},
+              {subdir_dep, "other_dep"}]}
+    ).
```

Comment:
Do we know what happens when we fetch two distinct subdir of the same base repo?